### ISSUE TITLE
Htools breaks when drbd disk template is not enabled

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#927]
Originally reported of Google Code with ID 871.

```
What software version are you running? Please provide the output of "gnt-
cluster --version", "gnt-cluster version", and "hspace --version".
<b>What distribution are you using?</b>

We run Ganeti 2.9.3 pulled from wheezy-backports with 2 patches (KVM boot from http and zsh completion). All the hosts run (fully-updated) Wheezy.

gnt-cluster version 
Software version: 2.9.3
Internode protocol: 2090000
Configuration format: 2090000
OS api version: 20
Export interface: 0
VCS version: v2.9.3

hspace --version
hspace (ganeti) version v2.9.3
compiled with ghc 7.4
running on linux x86_64

gnt-cluster --version 
gnt-cluster (ganeti v2.9.3) 2.9.3

<b>What steps will reproduce the problem?</b>
1. gnt-cluster modify --ipolicy-disk-templates=sharedfile
2. gnt-cluster modify --enabled-disk-templates=sharedfile
3. hbal -L -X

<b>What is the expected output? What do you see instead?</b>

hbal -L -X
Error: failed to load data, aborting: Node 'gnt3-01.gnt.grnet.gr', error while reading attribute 'spfree': Ganeti reports unavailable data


<b>Please provide any additional information below.</b>
We have clusters that use only sharedfile as a disk template. We decided to disable the rest of the disk templates so that we can minimize errors on instance creation. The change seems to break htools because they search for the spfree value.

```
Originally added on 2014-07-03 13:52:07 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-07-03 14:20:13 +0000 UTC.

Added Labels: Priority-High
Changed State: Accepted
Added to Milestone: Release2.9

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit 6e94c4eb0c90fa248736140f3c09b36b0536a4de
Author: Klaus Aehlig <aehlig@google.com>
Date:   Mon Jul 7 14:58:44 2014 +0200

    Make htools tolerate missing "spfree" on luxi
    
    If a cluster allows sharedfile as only disk template, the
    amount of free spindles might not be available. This is
    perfectly normal, hence the luxi backend of htools should
    handle this case gracefully and just report 0 available spindles.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Hrvoje Ribicic <riba@google.com>

```
Originally added on 2014-07-07 13:10:13 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
released as part of 2.11.6
```
Originally added on 2014-09-22 15:58:21 +0000 UTC.

Changed State: Released
